rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -833,12 +826,21 @@ void Player::configure_play_until_timestamp()
   if (play_options_.playback_duration >= rclcpp::Duration(0, 0) ||
     play_options_.playback_until_timestamp >= rcutils_time_point_value_t{0})
   {
-    play_until_timestamp_ = std::max(
-      starting_time_ + play_options_.playback_duration.nanoseconds(),
-      play_options_.playback_until_timestamp);
+    // Handling special case when playback_duration = 0
+    auto play_until_from_duration = (play_options_.playback_duration == rclcpp::Duration(0, 0)) ?
+      0 : starting_time_ + play_options_.playback_duration.nanoseconds();
+
+    play_until_timestamp_ =
+      std::max(play_until_from_duration, play_options_.playback_until_timestamp);
   } else {
     play_until_timestamp_ = -1;
   }
 }
 
+inline bool Player::shall_stop_at_timestamp(const rcutils_time_point_value_t & msg_timestamp) const
+{
+  if ((play_until_timestamp_ > -1 && msg_timestamp > play_until_timestamp_) ||
+    play_until_timestamp_ == 0) {return true;} else {return false;}
```

Comment:
I _think_ the [ROS 2 Style Guide](https://docs.ros.org/en/humble/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#open-versus-cuddled-braces) recommends the following:

```suggestion
  if (
    (play_until_timestamp_ > -1 && msg_timestamp > play_until_timestamp_) || play_until_timestamp_ == 0)
  {
    return true;
  } 
  else
  {
    return false;
  }
```

Reply:
@jhdcs Fixed. See new commit https://github.com/ros2/rosbag2/pull/1032/commits/cb8691ca1127de2f2589356d8d2a751d63335f3b